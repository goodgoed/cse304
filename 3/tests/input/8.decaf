class LocalScoping {
  void method() {
    int x;
    x = 10;
    {
      int y;
      y = x + 1;
    }
    int z;
    z = x + 2;
  }
}