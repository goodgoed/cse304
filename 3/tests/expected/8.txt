--------------------------------------------------------------------------
Class Name: LocalScoping
Superclass Name: 
Fields:
Constructors:
Methods:
METHOD: 1, method, LocalScoping, private, instance
Method Parameters: 
Variable Table:
VARIABLE: 1, x, local, int
VARIABLE: 3, z, local, int
Method Body:
Block([
Expr(Assign(Variable(1), Constant(Integer-constant(10)))),
Block([
Expr(Assign(Variable(2), Binary(add, Variable(1), Constant(Integer-constant(1)))))
]),
Expr(Assign(Variable(3), Binary(add, Variable(1), Constant(Integer-constant(2)))))
])
--------------------------------------------------------------------------
